zproject/test_extra_settings.py

```diff
@@ -173,8 +173,9 @@ def set_loglevel(logger_name: str, level: str) -> None:
 SOCIAL_AUTH_GOOGLE_SECRET = "secret"
 SOCIAL_AUTH_SUBDOMAIN = 'auth'
 SOCIAL_AUTH_APPLE_SERVICES_ID = 'com.zulip.chat'
-SOCIAL_AUTH_APPLE_BUNDLE_ID = 'com.zulip.bundle.id'
+SOCIAL_AUTH_APPLE_APP_ID = 'com.zulip.bundle.id'
 SOCIAL_AUTH_APPLE_CLIENT = 'com.zulip.chat'
+SOCIAL_AUTH_APPLE_AUDIENCE: List[Optional[str]] = [SOCIAL_AUTH_APPLE_APP_ID, SOCIAL_AUTH_APPLE_SERVICES_ID]
```

Comment:
Why `Optional`? Is `None` really a valid element of this list, or are you just missing a `SOCIAL_AUTH_APPLE_CLIENT is not None` check in `computed_settings`?

Reply:
mypy gets that `None` from having `SOCIAL_AUTH_APPLE_SERVICES_ID = get_secret('social_auth_apple_services_id', development_only=True)` [in](https://github.com/zulip/zulip/blob/111a1c60a931850248d71ba7e671080b344d18d6/zproject/default_settings.py#L79) `default_settings.py` where `get_secret` returns `Optional[str]`.

Reply:
Fixed with adding a `SOCIAL_AUTH_APPLE_CLIENT is not None` check.